### DISCUSSION TITLE
Handling multiple PxScene with single PxPhysics

### DISCUSSION THREAD

**Discussion by SumantBagri** [NVIDIA-Omniverse/PhysX#228] [Q&A]
Hello NVIDIA Devs,

Happy Holidays!!

I am currently working on a project where I am planning on creating multiple `PxScene` instances with a single `PxPhysics` instance. In this particular case, I would also like to enable the GPU rigid-body-pipeline on all the `PxScene` instances. I wanted to know the following:
1. If I have N \* X rigid bodies, is it better to have X objects distributed across N, GPU-enabled `PxScene` instances or have N \* X objects in a single GPU-enabled `PxScene`. Better in terms of GPU memory access patterns that would ultimately affect the performance
2. If there is a snippet that manages multiple GPU-enabled `PxScene` instances, could you point me to it ?
3. If there are any specific details that I should be aware of while maintaining multiple, GPU-enabled `PxScene` instances ?

Thank you in advance!

Hope you have an amazing holiday season :)

**Comment by preist-nvidia** [comment]
Hi @SumantBagri,

Happy holidays to you, too!

As so often, the answer is "it depends" on your specific simulation setup, and you should try both approaches and measure memory / perf and compare. But I asked the team and we would think that:
- In most cases you will be faster with a single scene for N*X rigids.
- The multi-scene approach will use more memory than the single-scene approach.

We don't have a multi-GPU-scene snippet, unfortunately. I am not aware of any special details you should be aware of for multi-scene GPU setups.

Please update us about what you find, or if you run into issues!

Cheers,
Philipp

Internal tracking: PX-4670
